src/planner/parabolic/ParabolicTimer.cpp

```diff
@@ -154,94 +102,12 @@ std::unique_ptr<trajectory::Spline> computeParabolicTiming(
-    stateSpace->logMap(relativeState, tangentVector);
-    milestones.emplace_back(toVector(tangentVector));
-  }
-
-  // Compute the timing of the path.
-  ParabolicRamp::DynamicPath dynamicPath;
-  // TODO: Should I apply the adjoint transform to these limits?
-  dynamicPath.Init(toVector(_maxVelocity), toVector(_maxAcceleration));
-  dynamicPath.SetMilestones(milestones);
-
-  // Construct a list of all ramp transition points.
-  double t = 0.;
-  std::set<double> transitionTimes;
-  transitionTimes.insert(t);
-
-  for (const auto& rampNd : dynamicPath.ramps)
-  {
-    for (const auto& ramp1d : rampNd.ramps)
-    {
-      transitionTimes.insert(t + ramp1d.tswitch1);
-      transitionTimes.insert(t + ramp1d.tswitch2);
-    }
-
-    t += rampNd.endTime;
-    transitionTimes.insert(t);
-  }
-
-  // Convert the output to a spline with a knot at each transition time.
-  assert(!transitionTimes.empty());
-  const auto startIt = std::begin(transitionTimes);
-  double timePrev = *startIt;
-  transitionTimes.erase(startIt);
-
-  Eigen::VectorXd positionPrev, velocityPrev;
-  evaluateAtTime(dynamicPath, timePrev, positionPrev, velocityPrev);
-
-  auto outputTrajectory = make_unique<trajectory::Spline>(
-      stateSpace, timePrev + _inputTrajectory.getStartTime());
-  auto segmentStartState = stateSpace->createState();
-
-  for (const auto timeCurr : transitionTimes)
-  {
-    Eigen::VectorXd positionCurr, velocityCurr;
-    evaluateAtTime(dynamicPath, timeCurr, positionCurr, velocityCurr);
-
-    // Compute the spline coefficients for this segment of the trajectory. Each
-    // segment is expressed in the tangent space of the previous segment.
-    // TODO: We should apply the adjoint transform to velocity.
-    CubicSplineProblem problem(Vector2d(0, timeCurr - timePrev), 4, dimension);
-    problem.addConstantConstraint(0, 0, Eigen::VectorXd::Zero(dimension));
-    problem.addConstantConstraint(0, 1, velocityPrev);
-    problem.addConstantConstraint(1, 0, positionCurr - positionPrev);
-    problem.addConstantConstraint(1, 1, velocityCurr);
-    const auto spline = problem.fit();
-
-    // Forward-integrate the trajectory from the start in one step. This should
-    // prevent accumulation of numerical error over long trajectories. Note
-    // that this is only possible in real vector spaces and SO(2).
-    stateSpace->expMap(positionPrev, relativeState);
-    stateSpace->compose(startState, relativeState, segmentStartState);
-
-    // Add the ramp to the output trajectory.
-    assert(spline.getCoefficients().size() == 1);
-    const auto& coefficients = spline.getCoefficients().front();
-    outputTrajectory->addSegment(
-        coefficients, timeCurr - timePrev, segmentStartState);
-
-    timePrev = timeCurr;
-    positionPrev = positionCurr;
-    velocityPrev = velocityCurr;
-  }
+  double startTime = _inputTrajectory.getStartTime();
+  auto dynamicPath = convertToDynamicPath(
+      _inputTrajectory, _maxVelocity, _maxAcceleration, false);
```

Comment:
@mkoval By what you suggested, here we should pass `_useVelocity = false`, right? Because we are converting an untimed piecewise linear `Spline` to a `DynamicPath` (after making changes by above suggestion).